File: 387-first-unique-character-in-a-string/first-unique-character-in-a-string.java
class Solution {
    public int firstUniqChar(String s) {

        int[] frequencies = new int[26];

        for(char c : s.toCharArray()){
            frequencies[c-'a']++;
        }

        for(int i=0; i<s.length(); i++){
          if(frequencies[s.charAt(i)-'a'] == 1) return i;
        }

        return -1;
        
    }
}

File: 387-first-unique-character-in-a-string/first-unique-character-in-a-string.py
class Solution(object):
    def firstUniqChar(self, word):
        freq_map = defaultdict(int)
        for c in word:
            freq_map[c] += 1
        
        for i in range(len(word)):
            if freq_map[word[i]] == 1:
                return i
        
        return -1
        